module/VuFind/src/VuFind/Db/Service/TagService.php

```diff
@@ -72,30 +74,32 @@ public function matchText(string $text): array
     /**
      * Get tags associated with the specified resource.
      *
-     * @param string $id          Record ID to look up
-     * @param string $source      Source of record to look up
-     * @param int    $limit       Max. number of tags to return (0 = no limit)
-     * @param ?int   $list        ID of list to load tags from (null for no
-     * restriction,  true for on ANY list, false for on NO list)
-     * @param ?int   $user        ID of user to load tags from (null for all users)
-     * @param string $sort        Sort type ('count' or 'tag')
-     * @param ?int   $userToCheck ID of user to check for ownership (this will
-     * not filter the result list, but rows owned by this user will have an is_me
-     * column set to 1)
+     * @param string                                $id        Record ID to look up
+     * @param string                                $source    Source of record to look up
+     * @param int                                   $limit     Max. number of tags to return (0 = no limit)
+     * @param UserListEntityInterface|int|bool|null $listOrId  List entity/ID to load tags from (null for no
+     * restriction, true for on ANY list, false for on NO lists)
+     * @param UserEntityInterface|int|null          $userOrId  User entity/ID to load tags from (null for all users)
+     * @param string                                $sort      Sort type ('count' or 'tag')
+     * @param UserEntityInterface|int|null          $ownerOrId Entity/ID representing user to check for ownership
+     * (this will not filter the result list, but rows owned by this user will have an is_me column set to 1)
      *
      * @return array
      */
     public function getForResource(
         string $id,
         string $source = DEFAULT_SEARCH_BACKEND,
         int $limit = 0,
-        ?int $list = null,
-        ?int $user = null,
+        UserListEntityInterface|int|bool|null $listOrId = null,
+        UserEntityInterface|int|null $userOrId = null,
         string $sort = 'count',
-        ?int $userToCheck = null
+        UserEntityInterface|int|null $ownerOrId = null
     ): array {
+        $listId = $listOrId instanceof UserListEntityInterface ? $listOrId->getId() : $listOrId;
```

Comment:
`$listIdOrBoolOrNull` (bad tongue-in-cheek suggestion, but the difficulty giving this a good name could be seen as an indication that it is a rather complex value).